sanitizer/_targets.js

```diff
@@ -15,52 +25,68 @@ function _setup( paramName, targetMap ) {
       // error & warning messages
       var messages = { errors: [], warnings: [] };
 
-      // the string of targets (comma delimeted)
+      // the string of targets (comma delimited)
       var targetsString = raw[opts.paramName];
 
-      // trim whitespace
-      if( _.isString( targetsString ) && !_.isEmpty( targetsString ) ){
-        targetsString = targetsString.trim();
-
-        // param must be a valid non-empty string
-        if( !_.isString( targetsString ) || _.isEmpty( targetsString ) ){
-          messages.errors.push(
-            opts.paramName + ' parameter cannot be an empty string. Valid options: ' + getValidKeys(opts.targetMap)
-          );
-        }
-        else {
-
-          // split string in to array and lowercase each target string
-          var targets = targetsString.split(',').map( function( target ){
-            return target.toLowerCase(); // lowercase inputs
-          });
-
-          // emit an error for each target *not* present in the targetMap
-          targets.filter( function( target ){
-            return !opts.targetMap.hasOwnProperty(target);
-          }).forEach( function( target ){
-            messages.errors.push(
-              '\'' + target + '\' is an invalid ' + opts.paramName + ' parameter. Valid options: ' + getValidKeys(opts.targetMap)
-            );
-          });
-
-          // only set types value when no error occured
-          if( !messages.errors.length ){
-            clean[opts.paramName] = targets.reduce(function(acc, target) {
-              return acc.concat(opts.targetMap[target]);
-            }, []);
-
-            // dedupe in case aliases expanded to common things or user typed in duplicates
-            clean[opts.paramName] = _.uniq(clean[opts.paramName]);
-          }
-        }
+      // input is not a string, parameter is not defined, can quit early
+      if (!_.isString(targetsString)) {
+        return messages;
       }
 
-      // string is empty
-      else if( _.isString( targetsString ) ){
+      // remove all whitespace characters
+      targetsString = targetsString.replace(/\s/g,'');
+
+      // return error if parameter ends up being all whitespace
+      if( _.isEmpty( targetsString ) ){
         messages.errors.push(
           opts.paramName + ' parameter cannot be an empty string. Valid options: ' + getValidKeys(opts.targetMap)
         );
+        return messages;
+      }
+
+      // split string in to array and lowercase each target string
+      var targets = targetsString.split(',').map( function( target ){
```

Comment:
style: use fat arrow function?

Reply:
Oh yeah, definitely. Makes this all fit on one line :)